File: bench/promptgreen/tasks/t13/py/hidden_tests.py
import os
import subprocess
import sys

HERE = os.path.dirname(os.path.abspath(__file__))
EXPECTED = "1013.7\n1027.58769\n1041.665641\n1055.93646\n1070.40279\n"


def main():
    result = subprocess.run(
        [sys.executable, os.path.join(HERE, "reference.py")],
        capture_output=True,
        text=True,
        check=False,
    )
    assert result.returncode == 0, (
        f"nonzero exit: {result.returncode}, stderr={result.stderr}"
    )
    assert result.stdout == EXPECTED, f"expected {EXPECTED!r}, got {result.stdout!r}"

    lines = [l for l in result.stdout.split("\n") if l]
    assert len(lines) == 5, f"expected 5 lines, got {len(lines)}"
    assert lines[1] == "1027.58769", f"period-2 edge mismatch: {lines[1]!r}"

    print("ok")


if __name__ == "__main__":
    main()

File: bench/promptgreen/tasks/t13/hidden_tests.mjs
// Hidden test for t13 (compound interest table). Never shown to an author.
const EXPECTED = '1013.7\n1027.58769\n1041.665641\n1055.93646\n1070.40279\n';

export async function run(compileFn, source) {
  const r = compileFn(source);
  const checks = [];

  checks.push({ name: 'exact-stdout', pass: r.ok === true && r.stdout === EXPECTED });

  const lines = (r.stdout || '').split('\n').filter((l) => l.length > 0);
  checks.push({ name: 'five-lines', pass: lines.length === 5 });

  // Edge: period 2 does not terminate at a clean digit count without the
  // odd basis-point rate (137) forcing extra fractional digits (5 digits:
  // 1027.58769); a naive truncation-only or floating-point implementation
  // tends to drift here.
  checks.push({ name: 'period-2-exact', pass: lines[1] === '1027.58769' });

  const green = checks.every((c) => c.pass);
  return { green, detail: { expected: EXPECTED, got: r.stdout, ok: r.ok, checks } };
}
